packages/devtools_app/lib/src/inspector/flutter/inspector_screen.dart

```diff
@@ -101,30 +115,39 @@ class _InspectorScreenBodyState extends State<InspectorScreenBody>
               child: Label(
                 FlutterIcons.refresh,
                 'Refresh Tree',
-                minIncludeTextWidth: 800,
+                minIncludeTextWidth: 900,
               ),
             ),
+            if (InspectorTreeControllerFlutter
+                .isExperimentalStoryOfLayoutEnabled)
+              Container(
+                margin: const EdgeInsets.only(left: 8.0),
+                child: OutlineButton(
+                  onPressed:
+                      summaryTreeController?.toggleDebugLayoutSummaryEnabled,
+                  child: const Label(
+                    FlutterIcons.lightbulb,
+                    'Show Constraints',
+                    minIncludeTextWidth: 1000,
+                  ),
+                ),
+              ),
             const Spacer(),
             Row(children: getServiceExtensionWidgets()),
           ],
         ),
         Expanded(
           child: Row(
             children: [
+              summaryTree,
               Expanded(
-                  child: InspectorTree(
-                controller: summaryTreeController,
-              )),
-              Expanded(
-                child: Stack(
-                  children: [
-                    InspectorTree(
-                      controller: detailsTreeController,
-                    ),
-                    if (_expandCollapseSupported) _expandCollapseButtons()
-                  ],
+                child: InspectorDetailsTabController(
+                  detailsTree: detailsTree,
+                  summaryTreeController: summaryTreeController,
+                  expandCollapseSupported: _expandCollapseSupported,
+                  expandCollapseButtons: _expandCollapseButtons(),
```

Comment:
A bit unsure whether I should build this button here and pass it down the tree, tried creating the button in the TabController instead but it is pretty hard since it depends on state attributes here

Reply:
Does expand-collapse make sense for the "Layout Details" tab or only for the "Details Tree" tab?

Reply:
its fine to pass some UI details down to lower parts of the tree. If you do need to, I recommend passing the data down in a generic manner.
For example, here what you really want is to indicate an extra widget that needs to be included next to the tab bar not what the widgets do.
For example, there might be additional extra widgets that would be useful here. So create an extra parameter on the InspectorDetailsTabController called 
`actionButtons` or something similar.

Reply:
for now expand-collapse buttons don't make sense when user is currently on layout details tab since we don't have anything to expand/collapse in the layout details tab, maybe I will disable this button in the next CL?

Reply:
we could also have them fade out if you happen to be in a tab that doesn't support them. I do like how you have put the buttons on the line with the tab bar.